src/main/scala/org/allenai/pnp/semparse/SemanticParser.scala

```diff
@@ -126,10 +127,14 @@ class SemanticParser(val actionSpace: ActionSpace, val vocab: IndexedList[String
         }
       }
       Preconditions.checkState(lastOutput != null)
-      
+      // Entity encoding is average of the representations of the corresponding span.
+      lastOutput = lastOutput / name.length
       val spanVector = new FloatVector(tokens.length)
       for (i <- 0 until tokens.length) {
-        val value = if (i >= span.start && i < span.end) { 1.0 } else { 0.0 }
+        val spanStart = if(span == null) -1 else span.start
+        val spanEnd = if(span == null)  -1 else span.end
```

Comment:
Also put the check for whether the span exists outside the loop. The logic here is confusing.